lib/elixir/lib/kernel.ex

```diff
@@ -1126,6 +1126,17 @@ defmodule Kernel do
     :erlang.spawn_monitor(module, fun, args)
   end
 
+  @doc """
+  Applies a function over a term and returns term
+
+  Useful for running side effects
+  """
+  @spec tap(x :: any, (x -> any)) :: x
```

Comment:
```suggestion
  @spec tap(x, (x -> any)) :: x when x: any
```

Reply:
Use “var” instead of any. Also please add examples, @doc since, and tests.